### ISSUE TITLE
Modifying GitHub actions for label change from size to level 

### ISSUE THREAD

**Issue by SAUMILDHANKAR** [hackforla/website#3109]
### Dependency
- [x] #2777 

### Overview
As a developer, I would like to track the size of an issue in terms of number of hours. In order to achieve that, we need to modify GitHub action automations or other files before changing size labels to level.

### Details
We are planning to edit the following labels:
- ```Size: Good second issue``` to ```level: Good second issue```
- ```Size: Small``` to ```level: Small```
- ```Size: Medium``` to ```level: Medium```
- ```Size: Large``` to ```level: Large```

However, before doing that we need to edit all the automations that might be dependent on these existing labels, so that no workflows are affected by this change. Refer to issue #2777 for details on which automations might be impacted.

### Action Items
- [x] Use the search string in the resources below to confirm that the list of files that use the size labels identified, is complete.
- [x] Refer to Resources section for which files will be impacted
- [x] Test the changes in your own repo and make sure other functionalities are not impacted
- [ ] Create a pull request if all looks good in your own repo
    - [ ] Note: Discuss a timeline with the merge team and leads to change the actual labels in our repo as this might be needed for the GitHub action to work properly.
- [x] **For merge team**: Once the pull request associated with this issue is approved and merged, please update and edit #3110 by
  - [x] Checking off the dependency for this issue in issue #3110
  - [x] Editing any action items if required
  - [x] If all dependencies are checked off, please move issue #3110  to the New Issue Approval column, remove the Dependency label and add ready for milestone label.

### Resources/Instructions
- [GHA architecture spreadsheet](https://docs.google.com/spreadsheets/d/12NcZQoyGYlHlMQtJE2IM8xLYpHN75agb/edit#gid=1231634015)
- [Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
- [Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
- https://github.com/hackforla/website/wiki/Hack-for-LA's-GitHub-Actions
- https://github.com/hackforla/website/wiki/HfLA-GitHub-Actions
- https://github.com/hackforla/website/wiki/GHA:-issue-trigger

#### Code that Reference Size Labels
- [Search GitHub Actions Workflow directory for files that contain the string `size:`](https://github.com/search?q=repo%3Ahackforla%2Fwebsite+size%3A+path%3A%2F%5Egithub-actions%5C%2F%2F&type=code&l=github-actions%2Ftrigger-issue%2Fadd-missing-labels-to-issues%2F)
- [GitHub actions folder](https://github.com/hackforla/website/tree/gh-pages/github-actions)
- [GitHub Actions Workflow Directory](https://github.com/hackforla/website/tree/gh-pages/.github/workflows)
1. [post-labels-comment.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-issue/add-missing-labels-to-issues/post-labels-comment.js) --> contains many actions that utilize labels using the 'size' attribute one time in the declaration
2. [check-labels.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-issue/add-missing-labels-to-issues/check-labels.js) ---> also contains processes that utilize labels using the 'size' attribute one time in the declaration

I will also note that the following phrases were not used throughout the 2 directories listed above. Just the use of "Size:" and labels.
- ```Size: Good second issue``` 
- ```Size: Small``` 
- ```Size: Medium``` 
- ```Size: Large``` 

#### Wiki pages that reference size labels
- [Search the wiki for "size: medium"](https://github.com/search?q=repo%3Ahackforla%2Fwebsite+size%3A+path%3A%2F%5Egithub-actions%5C%2F%2F&type=code&l=github-actions%2Ftrigger-issue%2Fadd-missing-labels-to-issues%2F)
1. https://github.com/hackforla/website/wiki/How-to-read-and-interpret-labels
2. https://github.com/hackforla/website/wiki/How-to-Review-Pull-Requests
3. https://github.com/hackforla/website/wiki/How-to-create-issues

**Comment by KazushiR** [comment]
Progress: self-assigned
Availability: 6 hrs for the rest of this week
ETA: 06/17/2022

**Comment by bishrfaisal** [comment]
Good work writing this issue! I have one suggestion that we could discuss during today's team meeting. Would it be possible for Github automation to only detect the keywords of the size, such as:
- "Large"
- "Medium"
- "Small"
- "Good second issue"
- "good first issue"
- "Third Issue"

We would like this to be irrespective of capitalization and whatever word comes before such as:
- "Size"
- "Complexity"
- "Order"
- "Fun"

This is intended to future-proof this in case a more desirable name could be added before the size.

**Comment by jdingeman** [comment]
Here are some relevant links so we can discuss this issue later:

- https://github.com/hackforla/website/wiki/Hack-for-LA's-GitHub-Actions
- https://github.com/hackforla/website/wiki/HfLA-GitHub-Actions
- https://github.com/hackforla/website/wiki/GHA:-issue-trigger

**Comment by 7kram** [comment]
It was determined that this is not feasible because of "size: missing"

- [ ] Research if it would be possible for Github automation to only detect the keywords of the size, such as:
          - "Large"
          - "Medium"
          - "Small"
          - "Good second issue"
          - "good first issue"
          - "Third Issue"
        We would like this to be irrespective of capitalization and whatever word comes before such as:
          - "Size"
          - "level"
          - "Order"
          - "Fun"
        This is intended to future-proof this in case a more desirable name could be added before the size.

**Comment by bzzz-coding** [comment]
i. Availability: Available 1-2 hours, M-Th evenings 
ii. ETA: 1/25/23

**Comment by bzzz-coding** [comment]
Files that reference size labels:
- github-actions/trigger-issue/add-missing-labels-to-issues/check-labels.js
- github-actions/trigger-issue/add-missing-labels-to-issues/post-labels-comment.js
- CONTRIBUTING.md


Wiki pages that reference size labels:
- https://github.com/hackforla/website/wiki/How-to-create-issues
- https://github.com/hackforla/website/wiki/How-to-read-and-interpret-labels#the-size-series

**Comment by bzzz-coding** [comment]
Update
1. Progress:
    ## Progress 1: Identified files and pages where issue Size is referenced:
    ### Files that reference size labels:
       - github-actions/trigger-issue/add-missing-labels-to-issues/check-labels.js
       - github-actions/trigger-issue/add-missing-labels-to-issues/post-labels-comment.js
       - CONTRIBUTING.md
    ### Wiki pages that reference size labels:
       - https://github.com/hackforla/website/wiki/How-to-create-issues
       - https://github.com/hackforla/website/wiki/How-to-read-and-interpret-labels#the-size-series
      
    ## Progress 2: Changed 'Size' to 'level' in the two files in github-actions:
       https://github.com/hackforla/website/compare/gh-pages...bzzz-coding:HfLA-website:modify-github-actions-for-label-change-3109

3. Blockers: I'm not sure how to "test the changes in your own repo and make sure other functionalities are not impacted"
4. Availability: Available 1-2 hours, M-Th evenings; 1 hour, Friday morning
5. ETA: 1/25/23

**Comment by bzzz-coding** [comment]
## Update regarding testing:
- Made a pull request from my feature branch to my forked gh-pages default branch
- Merged the changes (updating Size to level in github-actions files)
- Tried opening an issue without adding labels and GHA returned level:missing as predicted
<img width="1276" alt="Screen Shot 2023-01-13 at 11 22 27 AM" src="https://user-images.githubusercontent.com/86077274/212405819-b24dd351-6c42-4839-bdb3-19fa3025cfdc.png">


## However, since the actual labels 'Size: Large' need to be edited in the original repo, I can't do further testing in my own repo.

<img width="1470" alt="Screen Shot 2023-01-13 at 11 43 01 AM" src="https://user-images.githubusercontent.com/86077274/212405769-691b949e-788f-4a00-879d-a471188dd3ac.png">

## Next steps:

- [ ] Update CONTRIBUTING.md -- I'm not sure if I'm supposed to do this as part of this issue.
- [ ] Update Wiki pages:
    - [ ] - https://github.com/hackforla/website/wiki/How-to-create-issues
    - [ ]  https://github.com/hackforla/website/wiki/How-to-read-and-interpret-labels#the-size-series
- [ ] Update Size labels and (maybe) delete 'size: missing' since we have 'level: missing' now https://github.com/hackforla/website/labels?q=size

**Comment by MattPereira** [comment]
-  #3921 

The `Check Labels` step of the `Add-Missing-Labels-To-Issues` is functioning as expected by adding the missing labels, however, I noticed that the `Post Comment` step of the job is being skipped because the `checkUserMember` step that precedes posting a comment is failing with `Error: Bad credentials`

- [Test Issue GHA logs](https://github.com/hackforla/website/actions/runs/4068731308/jobs/7007633243) 
- [issue-trigger.yml](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/issue-trigger.yml)

**Comment by MattPereira** [comment]
Also, why did the `HackforLABot` decide to move this issue from QA to Done? That seems incorrect

**Comment by JessicaLucindaCheng** [comment]
> * [Testing if GHA that adds missing labels is updated from "Size" to "Complexity #3921](https://github.com/hackforla/website/issues/3921)
> 
> The `Check Labels` step of the `Add-Missing-Labels-To-Issues` is functioning as expected by adding the missing labels, however, I noticed that the `Post Comment` step of the job is being skipped because the `checkUserMember` step that precedes posting a comment is failing with `Error: Bad credentials`
> 
> * [Test Issue GHA logs](https://github.com/hackforla/website/actions/runs/4068731308/jobs/7007633243)
> * [issue-trigger.yml](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/issue-trigger.yml)

I think this is being addressed in issue #4189




> Also, why did the `HackforLABot` decide to move this issue from QA to Done? That seems incorrect

I think (though I might be wrong) the `HackforLABot` moves any back end issues to Done because Bonnie generally can't visually inspect and QA back end issues anyway.
